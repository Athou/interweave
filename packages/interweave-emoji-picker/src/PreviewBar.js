/**
 * @copyright   2016, Miles Johnson
 * @license     https://opensource.org/licenses/MIT
 * @flow
 */

import React from 'react';
import PropTypes from 'prop-types';
import EmojiCharacter from 'interweave/lib/components/Emoji';
import { EmojiShape, EmojiPathShape } from './shapes';

import type { Emoji, EmojiPath } from './types';

type PreviewBarProps = {
  emoji: ?Emoji,
  emojiPath: EmojiPath,
  showShortcodes: boolean,
};

export default class PreviewBar extends React.PureComponent<PreviewBarProps> {
  static contextTypes = {
    messages: PropTypes.objectOf(PropTypes.string),
  };

  static propTypes = {
    emoji: EmojiShape,
    emojiPath: EmojiPathShape.isRequired,
    showShortcodes: PropTypes.bool.isRequired,
  };

  static defaultProps = {
    emoji: null,
  };

  formatAnnotation(annotation: string): string {
    return annotation
      .split(' ')
      .map(word => word.charAt(0).toUpperCase() + word.slice(1))
      .join(' ');
  }

  formatShortcodes(shortcodes: string[]): string {
    return shortcodes.map(code => `:${code}:`).join(' ');
  }

  render() {
    const { emoji, emojiPath, showShortcodes } = this.props;

    if (!emoji) {
      return (
        <div className="iep__preview iep__preview--empty">
          {this.context.messages['no-preview']}
        </div>
      );
    }

    return (
      <div className="iep__preview">
        <div className="iep__preview-emoji">
          <EmojiCharacter
            unicode={emoji.emoji || emoji.text}
            emojiPath={emojiPath}
            emojiSize={2.5}
          />
        </div>

        <div className="iep__preview-content">
          {emoji.annotation && (
            <div className="iep__preview-name">
              {this.formatAnnotation(emoji.annotation)}
            </div>
          )}

          {showShortcodes && emoji.shortcodes && (
            <div className="iep__preview-shortcodes">
              {this.formatShortcodes(emoji.shortcodes)}
            </div>
          )}
        </div>
      </div>
    );
  }
}
