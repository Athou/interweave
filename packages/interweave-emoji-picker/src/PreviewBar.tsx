/**
 * @copyright   2016-2018, Miles Johnson
 * @license     https://opensource.org/licenses/MIT
 */

import React from 'react';
import PropTypes from 'prop-types';
import EmojiCharacter, {
  CanonicalEmoji,
  EmojiShape,
  Path,
  PathShape,
  Size,
  SizeShape,
  Source,
  SourceShape,
} from 'interweave-emoji';
import withContext, { ContextProps } from './Context';
import { ContextShape } from './shapes';

export interface PreviewBarProps {
  emoji?: CanonicalEmoji | null;
  emojiLargeSize: Size;
  emojiPath: Path;
  emojiSource: Source;
  hideEmoticon: boolean;
  hideShortcodes: boolean;
}

const TITLE_REGEX: RegExp = /(^|:|\.)\s?[a-z]/g;

export class PreviewBar extends React.PureComponent<PreviewBarProps & ContextProps> {
  static propTypes = {
    context: ContextShape.isRequired,
    emoji: EmojiShape,
    emojiLargeSize: SizeShape.isRequired,
    emojiPath: PathShape.isRequired,
    emojiSource: SourceShape.isRequired,
    hideEmoticon: PropTypes.bool.isRequired,
    hideShortcodes: PropTypes.bool.isRequired,
  };

  static defaultProps = {
    emoji: null,
  };

  /**
   * Format the title using sentence case.
   */
  formatTitle(title: string): string {
    const { emoji } = this.props;

    if (!emoji) {
      return '';
    }

    const { annotation, primary_shortcode: shortcode = '' } = emoji;

    // Flag and country names are cased correctly
    if (annotation && shortcode.indexOf('flag_') >= 0) {
      return annotation;
    }

    // Uppercase certain characters
    return title.replace(TITLE_REGEX, token => token.toUpperCase());
  }

  render() {
    const {
      context: { classNames, messages },
      emoji,
      emojiLargeSize,
      emojiPath,
      emojiSource,
      hideEmoticon,
      hideShortcodes,
    } = this.props;

    if (!emoji) {
      return (
        <div className={classNames.preview}>
          {messages.noPreview && <div className={classNames.noPreview}>{messages.noPreview}</div>}
        </div>
      );
    }

    const title = emoji.annotation || emoji.name;
    const subtitle = [];

    if (!hideEmoticon && emoji.emoticon) {
      subtitle.push(emoji.emoticon);
    }

    if (!hideShortcodes && emoji.canonical_shortcodes) {
      subtitle.push(...emoji.canonical_shortcodes);
    }

    return (
      <div className={classNames.preview}>
        <div className={classNames.previewEmoji}>
          <EmojiCharacter
            emojiLargeSize={emojiLargeSize}
            emojiPath={emojiPath}
            emojiSource={emojiSource}
            enlargeEmoji
            hexcode={emoji.hexcode}
          />
        </div>

        <div className={classNames.previewContent}>
          {title && <div className={classNames.previewTitle}>{this.formatTitle(title)}</div>}

          {subtitle.length > 0 && (
            <div className={classNames.previewSubtitle}>{subtitle.join(' ')}</div>
          )}
        </div>
      </div>
    );
  }
}

export default withContext(PreviewBar);
