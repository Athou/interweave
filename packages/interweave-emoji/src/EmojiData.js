/**
 * @copyright   2016, Miles Johnson
 * @license     https://opensource.org/licenses/MIT
 * @flow
 */

/* eslint-disable no-param-reassign */

import {
  fromCodepointToUnicode,
  fromHexcodeToCodepoint,
  generateEmoticonPermutations,
} from 'emojibase';
import { TEXT, EMOTICON_OPTIONS } from 'emojibase/lib/constants';

import type { Emoji } from 'emojibase';

const instances: { [locale: string]: EmojiData } = {}; // eslint-disable-line

export default class EmojiData {
  EMOJIS: { [hexcode: string]: Emoji } = {};
  EMOTICON_TO_HEXCODE: { [emoticon: string]: string } = {};
  SHORTCODE_TO_HEXCODE: { [shortcode: string]: string } = {};
  UNICODE_TO_HEXCODE: { [unicode: string]: string } = {};

  data: Emoji[] = [];
  flatData: Emoji[] = [];
  locale: string = 'en';

  constructor(locale: string) {
    this.locale = locale;
  }

  /**
   * Return or create a singleton instance per locale.
   */
  static getInstance(locale: string) {
    if (!instances[locale]) {
      instances[locale] = new EmojiData(locale);
    }

    return instances[locale];
  }

  /**
   * Return dataset as a list.
   */
  getData(): Emoji[] {
    return this.data;
  }

  /**
   * Return dataset as a flattened list.
   */
  getFlatData(): Emoji[] {
    return this.flatData;
  }

  /**
   * Package the emoji object with additional data,
   * while also extracting and partitioning relevant information.
   */
  packageEmoji(emoji: Object): Emoji {
    const { emoticon, hexcode, shortcodes } = emoji;

    // Make our lives easier
    if (!emoji.unicode) {
      emoji.unicode = (emoji.text && emoji.type === TEXT) ? emoji.text : emoji.emoji;
    }

    // Canonicalize the shortcodes for easy reuse
    emoji.canonical_shortcodes = shortcodes.map(code => `:${code}:`);
    emoji.primary_shortcode = emoji.canonical_shortcodes[0]; // eslint-disable-line

    // Support all shortcodes
    emoji.canonical_shortcodes.forEach((shortcode) => {
      this.SHORTCODE_TO_HEXCODE[shortcode] = hexcode;
    });

    // Support all emoticons
    if (emoticon) {
      generateEmoticonPermutations(emoticon, EMOTICON_OPTIONS[emoticon]).forEach((emo) => {
        this.EMOTICON_TO_HEXCODE[emo] = hexcode;
      });
    }

    // Support all presentations (even no variation selectors)
    this.UNICODE_TO_HEXCODE[fromCodepointToUnicode(fromHexcodeToCodepoint(hexcode))] = hexcode;

    if (emoji.emoji) {
      this.UNICODE_TO_HEXCODE[emoji.emoji] = hexcode;
    }

    if (emoji.text) {
      this.UNICODE_TO_HEXCODE[emoji.text] = hexcode;
    }

    // Map each emoji
    this.EMOJIS[hexcode] = emoji;

    return emoji;
  }

  /**
   * Parse and generate emoji datasets.
   */
  parseEmojiData(data: Emoji[]): Emoji[] {
    data.forEach((emoji) => {
      const packagedEmoji = this.packageEmoji(emoji);

      this.data.push(packagedEmoji);
      this.flatData.push(packagedEmoji);

      // Flatten and package skins as well
      if (packagedEmoji.skins) {
        packagedEmoji.skins.forEach((skin) => {
          this.flatData.push(this.packageEmoji(skin));
        });
      }
    });

    return this.data;
  }
}
