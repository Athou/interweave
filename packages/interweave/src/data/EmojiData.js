/**
 * @copyright   2016, Miles Johnson
 * @license     https://opensource.org/licenses/MIT
 * @flow
 */

/* eslint-disable no-param-reassign */

import { generateEmoticonPermutations } from 'emojibase';
import { TEXT, EMOTICON_OPTIONS } from 'emojibase/lib/constants';

import type { Emoji } from 'emojibase';

const instances: { [locale: string]: EmojiData } = {}; // eslint-disable-line

export default class EmojiData {
  EMOJIS: { [unicode: string]: Emoji } = {};
  EMOTICON_TO_UNICODE: { [emoticon: string]: string } = {};
  SHORTCODE_TO_UNICODE: { [shortcode: string]: string } = {};
  UNICODE_TO_SHORTCODES: { [unicode: string]: string[] } = {};

  data: Emoji[] = [];
  flatData: Emoji[] = [];
  locale: string = 'en';

  constructor(locale: string) {
    this.locale = locale;
  }

  static getInstance(locale: string) {
    if (!instances[locale]) {
      instances[locale] = new EmojiData(locale);
    }

    return instances[locale];
  }

  getData(): Emoji[] {
    return this.data;
  }

  getFlatData(): Emoji[] {
    return this.flatData;
  }

  packageEmoji(emoji: Object): Emoji {
    const { emoticon, shortcodes } = emoji;

    // Only support the default presentation
    const unicode = (emoji.text && emoji.type === TEXT) ? emoji.text : emoji.emoji;
    emoji.unicode = unicode;

    // Canonicalize the shortcodes for easy reuse
    emoji.canonical_shortcodes = shortcodes.map(code => `:${code}:`);
    emoji.primary_shortcode = emoji.canonical_shortcodes[0]; // eslint-disable-line

    // Support all shortcodes
    this.UNICODE_TO_SHORTCODES[unicode] = emoji.canonical_shortcodes.map((shortcode) => {
      this.SHORTCODE_TO_UNICODE[shortcode] = unicode;

      return shortcode;
    });

    // Support all emoticons
    if (emoticon) {
      generateEmoticonPermutations(emoticon, EMOTICON_OPTIONS[emoticon]).forEach((emo) => {
        this.EMOTICON_TO_UNICODE[emo] = unicode;
      });
    }

    // Map each emoji by unicode
    this.EMOJIS[unicode] = emoji;

    return emoji;
  }

  parseEmojiData(data: Emoji[]): Emoji[] {
    data.forEach((emoji) => {
      const packagedEmoji = this.packageEmoji(emoji);

      this.data.push(packagedEmoji);
      this.flatData.push(packagedEmoji);

      // Flatten and package skins as well
      if (packagedEmoji.skins) {
        packagedEmoji.skins.forEach((skin) => {
          this.flatData.push(this.packageEmoji(skin));
        });
      }
    });

    return this.data;
  }
}
