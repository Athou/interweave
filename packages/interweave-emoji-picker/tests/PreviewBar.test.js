import React from 'react';
import { shallow } from 'enzyme';
import EmojiData from '../../interweave-emoji/src/EmojiDataManager';
import PreviewBar from '../src/PreviewBar';
import { SOURCE_PROP } from '../../../tests/mocks';
import { PICKER_CONTEXT } from './mocks';

describe('<PreviewBar />', () => {
  const context = PICKER_CONTEXT;

  const props = {
    emoji: null,
    emojiLargeSize: '3em',
    emojiPath: '{{hexcode}}',
    emojiSource: SOURCE_PROP,
    hideEmoticon: false,
    hideShortcodes: false,
  };

  beforeEach(() => {
    props.emoji = EmojiData.getInstance('en').EMOJIS['1F642'];
  });

  it('renders a preview', () => {
    const wrapper = shallow(<PreviewBar {...props} />, { context });

    expect(wrapper).toMatchSnapshot();
  });

  describe('with emoji', () => {
    it('can customize class name', () => {
      const wrapper = shallow(<PreviewBar {...props} />, {
        context: {
          ...context,
          classNames: {
            ...context.classNames,
            preview: 'test-preview',
            previewEmoji: 'test-preview-emoji',
            previewContent: 'test-preview-content',
            previewTitle: 'test-preview-title',
            previewSubtitle: 'test-preview-subtitle',
          },
        },
      });
      const divs = wrapper.find('div');

      expect(divs.at(0).prop('className')).toBe('test-preview');
      expect(divs.at(1).prop('className')).toBe('test-preview-emoji');
      expect(divs.at(2).prop('className')).toBe('test-preview-content');
      expect(divs.at(3).prop('className')).toBe('test-preview-title');
      expect(divs.at(4).prop('className')).toBe('test-preview-subtitle');
    });

    it('renders an emoji', () => {
      const wrapper = shallow(<PreviewBar {...props} />, { context });

      expect(wrapper.find('Emoji').props()).toEqual(
        expect.objectContaining({
          emojiLargeSize: props.emojiLargeSize,
          emojiPath: props.emojiPath,
          emojiSource: props.emojiSource,
          enlargeEmoji: true,
          hexcode: props.emoji.hexcode,
        }),
      );
    });

    it('displays title', () => {
      const wrapper = shallow(<PreviewBar {...props} />, { context });

      expect(wrapper.find('.interweave-picker__preview-title').text()).toBe(
        'Slightly smiling face',
      );
    });

    it('displays title when no annotation', () => {
      const wrapper = shallow(
        <PreviewBar
          {...props}
          emoji={{
            ...props.emoji,
            annotation: '',
          }}
        />,
        { context },
      );

      expect(wrapper.find('.interweave-picker__preview-title').text()).toBe(
        'SLIGHTLY SMILING FACE',
      );
    });

    it('doesnt display title if no annotation or name', () => {
      const wrapper = shallow(
        <PreviewBar
          {...props}
          emoji={{
            ...props.emoji,
            annotation: '',
            name: '',
          }}
        />,
        { context },
      );

      expect(wrapper.find('.interweave-picker__preview-title')).toHaveLength(0);
    });

    it('displays subtitle', () => {
      const wrapper = shallow(<PreviewBar {...props} />, { context });

      expect(wrapper.find('.interweave-picker__preview-subtitle').text()).toBe(':) :pleased:');
    });

    it('displays subtitle without emoticon', () => {
      const wrapper = shallow(
        <PreviewBar
          {...props}
          emoji={{
            ...props.emoji,
            emoticon: '',
          }}
        />,
        { context },
      );

      expect(wrapper.find('.interweave-picker__preview-subtitle').text()).toBe(':pleased:');
    });

    it('displays subtitle without shortcodes', () => {
      const wrapper = shallow(
        <PreviewBar
          {...props}
          emoji={{
            ...props.emoji,
            canonical_shortcodes: [],
          }}
        />,
        { context },
      );

      expect(wrapper.find('.interweave-picker__preview-subtitle').text()).toBe(':)');
    });

    it('hides emoticon in subtitle', () => {
      const wrapper = shallow(<PreviewBar {...props} hideEmoticon />, { context });

      expect(wrapper.find('.interweave-picker__preview-subtitle').text()).toBe(':pleased:');
    });

    it('hides shortcodes in subtitle', () => {
      const wrapper = shallow(<PreviewBar {...props} hideShortcodes />, { context });

      expect(wrapper.find('.interweave-picker__preview-subtitle').text()).toBe(':)');
    });

    it('doesnt display subtitle if no emoticon or shortcodes', () => {
      const wrapper = shallow(
        <PreviewBar
          {...props}
          emoji={{
            ...props.emoji,
            canonical_shortcodes: [],
            emoticon: '',
          }}
        />,
        { context },
      );

      expect(wrapper.find('.interweave-picker__preview-subtitle')).toHaveLength(0);
    });

    it('doesnt display subtitle if all are hidden', () => {
      const wrapper = shallow(<PreviewBar {...props} hideEmoticon hideShortcodes />, { context });

      expect(wrapper.find('.interweave-picker__preview-subtitle')).toHaveLength(0);
    });
  });

  describe('no emoji', () => {
    it('can customize class name', () => {
      const wrapper = shallow(<PreviewBar {...props} emoji={null} />, {
        context: {
          ...context,
          classNames: {
            ...context.classNames,
            preview: 'test-preview',
            noPreview: 'test-preview-none',
          },
        },
      });
      const divs = wrapper.find('div');

      expect(divs.at(0).prop('className')).toBe('test-preview');
      expect(divs.at(1).prop('className')).toBe('test-preview-none');
    });

    it('renders no preview message', () => {
      const none = <b>No preview</b>;
      const wrapper = shallow(<PreviewBar {...props} emoji={null} />, {
        context: {
          ...context,
          messages: {
            ...context.messages,
            noPreview: none,
          },
        },
      });

      expect(wrapper.find('div')).toHaveLength(2);
      expect(wrapper.contains(none)).toBe(true);
    });

    it('renders nothing if no preview message', () => {
      const wrapper = shallow(<PreviewBar {...props} emoji={null} />, {
        context: {
          ...context,
          messages: {
            ...context.messages,
            noPreview: '',
          },
        },
      });

      expect(wrapper.find('div')).toHaveLength(1);
    });
  });
});
