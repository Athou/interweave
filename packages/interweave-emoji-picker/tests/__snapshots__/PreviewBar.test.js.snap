// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PreviewBar /> renders a preview 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PreviewBar
    emoji={
      Object {
        "annotation": "slightly smiling face",
        "canonical_shortcodes": Array [
          ":pleased:",
        ],
        "emoji": "ðŸ™‚",
        "emoticon": ":)",
        "group": 0,
        "hexcode": "1F642",
        "name": "SLIGHTLY SMILING FACE",
        "order": 20,
        "primary_shortcode": ":pleased:",
        "shortcodes": Array [
          "pleased",
        ],
        "subgroup": 0,
        "tags": Array [
          "face",
          "smile",
        ],
        "type": 1,
        "unicode": "ðŸ™‚",
        "version": 1,
      }
    }
    emojiLargeSize="3em"
    emojiPath="{{hexcode}}"
    emojiSource={
      Object {
        "compact": false,
        "locale": "en",
        "version": "latest",
      }
    }
    hideEmoticon={false}
    hideShortcodes={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="interweave-picker__preview-emoji"
        >
          <Emoji
            emojiLargeSize="3em"
            emojiPath="{{hexcode}}"
            emojiSize="1em"
            emojiSource={
              Object {
                "compact": false,
                "locale": "en",
                "version": "latest",
              }
            }
            emoticon=""
            enlargeEmoji={true}
            hexcode="1F642"
            renderUnicode={false}
            shortcode=""
            unicode=""
          />
        </div>,
        <div
          className="interweave-picker__preview-content"
        >
          <div
            className="interweave-picker__preview-title"
          >
            Slightly smiling face
          </div>
          <div
            className="interweave-picker__preview-subtitle"
          >
            :) :pleased:
          </div>
        </div>,
      ],
      "className": "interweave-picker__preview",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Emoji
            emojiLargeSize="3em"
            emojiPath="{{hexcode}}"
            emojiSize="1em"
            emojiSource={
              Object {
                "compact": false,
                "locale": "en",
                "version": "latest",
              }
            }
            emoticon=""
            enlargeEmoji={true}
            hexcode="1F642"
            renderUnicode={false}
            shortcode=""
            unicode=""
          />,
          "className": "interweave-picker__preview-emoji",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "emojiLargeSize": "3em",
            "emojiPath": "{{hexcode}}",
            "emojiSize": "1em",
            "emojiSource": Object {
              "compact": false,
              "locale": "en",
              "version": "latest",
            },
            "emoticon": "",
            "enlargeEmoji": true,
            "hexcode": "1F642",
            "renderUnicode": false,
            "shortcode": "",
            "unicode": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="interweave-picker__preview-title"
            >
              Slightly smiling face
            </div>,
            <div
              className="interweave-picker__preview-subtitle"
            >
              :) :pleased:
            </div>,
          ],
          "className": "interweave-picker__preview-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Slightly smiling face",
              "className": "interweave-picker__preview-title",
            },
            "ref": null,
            "rendered": "Slightly smiling face",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": ":) :pleased:",
              "className": "interweave-picker__preview-subtitle",
            },
            "ref": null,
            "rendered": ":) :pleased:",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="interweave-picker__preview-emoji"
          >
            <Emoji
              emojiLargeSize="3em"
              emojiPath="{{hexcode}}"
              emojiSize="1em"
              emojiSource={
                Object {
                  "compact": false,
                  "locale": "en",
                  "version": "latest",
                }
              }
              emoticon=""
              enlargeEmoji={true}
              hexcode="1F642"
              renderUnicode={false}
              shortcode=""
              unicode=""
            />
          </div>,
          <div
            className="interweave-picker__preview-content"
          >
            <div
              className="interweave-picker__preview-title"
            >
              Slightly smiling face
            </div>
            <div
              className="interweave-picker__preview-subtitle"
            >
              :) :pleased:
            </div>
          </div>,
        ],
        "className": "interweave-picker__preview",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Emoji
              emojiLargeSize="3em"
              emojiPath="{{hexcode}}"
              emojiSize="1em"
              emojiSource={
                Object {
                  "compact": false,
                  "locale": "en",
                  "version": "latest",
                }
              }
              emoticon=""
              enlargeEmoji={true}
              hexcode="1F642"
              renderUnicode={false}
              shortcode=""
              unicode=""
            />,
            "className": "interweave-picker__preview-emoji",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "emojiLargeSize": "3em",
              "emojiPath": "{{hexcode}}",
              "emojiSize": "1em",
              "emojiSource": Object {
                "compact": false,
                "locale": "en",
                "version": "latest",
              },
              "emoticon": "",
              "enlargeEmoji": true,
              "hexcode": "1F642",
              "renderUnicode": false,
              "shortcode": "",
              "unicode": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="interweave-picker__preview-title"
              >
                Slightly smiling face
              </div>,
              <div
                className="interweave-picker__preview-subtitle"
              >
                :) :pleased:
              </div>,
            ],
            "className": "interweave-picker__preview-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Slightly smiling face",
                "className": "interweave-picker__preview-title",
              },
              "ref": null,
              "rendered": "Slightly smiling face",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": ":) :pleased:",
                "className": "interweave-picker__preview-subtitle",
              },
              "ref": null,
              "rendered": ":) :pleased:",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "classNames": Object {
        "emoji": "interweave-picker__emoji",
        "emojiActive": "interweave-picker__emoji--active",
        "emojis": "interweave-picker__emojis",
        "emojisBody": "interweave-picker__emojis-body",
        "emojisContainer": "interweave-picker__emojis-container",
        "emojisHeader": "interweave-picker__emojis-header",
        "emojisRow": "interweave-picker__emojis-row",
        "emojisSection": "interweave-picker__emojis-section",
        "group": "interweave-picker__group",
        "groupActive": "interweave-picker__group--active",
        "groups": "interweave-picker__groups",
        "groupsList": "interweave-picker__groups-list",
        "noPreview": "interweave-picker__no-preview",
        "noResults": "interweave-picker__no-results",
        "picker": "interweave-picker__picker",
        "pickerVirtual": "interweave-picker__picker--virtual",
        "preview": "interweave-picker__preview",
        "previewContent": "interweave-picker__preview-content",
        "previewEmoji": "interweave-picker__preview-emoji",
        "previewSubtitle": "interweave-picker__preview-subtitle",
        "previewTitle": "interweave-picker__preview-title",
        "search": "interweave-picker__search",
        "searchInput": "interweave-picker__search-input",
        "skinTone": "interweave-picker__skin-tone",
        "skinToneActive": "interweave-picker__skin-tone--active",
        "skinTones": "interweave-picker__skin-tones",
      },
      "messages": Object {
        "activities": "Activities",
        "animalsNature": "Animals & Nature",
        "flags": "Flags",
        "foodDrink": "Food & Drink",
        "frequentlyUsed": "Frequently Used",
        "noGroups": "All Emojis",
        "noPreview": "No emoji to display",
        "noResults": "No results",
        "objects": "Objects",
        "recentlyUsed": "Recently Used",
        "search": "Search",
        "searchA11y": "Search for emojis by keyword",
        "searchResults": "Search Results",
        "smileysPeople": "Smileys & People",
        "symbols": "Symbols",
        "travelPlaces": "Travel & Places",
      },
    },
  },
}
`;
